src/commissaire_http/authentication/keystonepasswordauth.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) 2016  Red Hat, Inc
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+OpenStack Keystone authentication plugin.
+"""
+
+import json
+import requests
+
+from commissaire_http.authentication import Authenticator
+from commissaire_http.authentication import decode_basic_auth
+
+
+class KeystonePassword(Authenticator):
+    """
+    Auth implementation using password method against OpenStack Keystone
+    """
+
+    def __init__(self, app, url=None, domain='Default'):
+        """
+        Checks the user name and password from an Authorization header
+        against the specified OpenStack Keystone instance
+
+        :param app: The WSGI application being wrapped with authenticaiton.
+        :type app: callable
+        :param url: The OpenStack Keystone endpoint used for Authentication
+        :type url: string
+        :returns: HTTPBasicAuth
+        """
+        super(KeystonePassword, self).__init__(app)
```

Comment:
Another nice Python 3 feature: ``super`` doesn't require input unless there is something special needed. This line could read: ``super().__init__(app)``. Not worth changing as this works totally fine just want to spread the Python 3 happiness :smile:.